### PR TITLE
1138 param_value_string bug fix

### LINKED ISSUE #1138: utils param_value_string missing struct array
`catena::param_value_string` in `utils.cpp` does not handle `kStructArrayValues` should output `[struct array values]`
Also does not handle `kUndefinedValue` or `KIND_NOT_SET` but those are probably less important.

### PR DESCRIPTION
**Pull request by KeonF-ross**
## 📖 Description

Addresses issue #1138 by modifying `catena::param_value_string`  in `utils.cpp` to handle the following value types: `kStructArrayValues`, `kUndefinedValue`, `KIND_NOT_SET`.

---

## ✅ Definition of Done (DoD)
- [x] Bug is fixed and feature works as expected
- [x] Edge cases are handled
- [x] Tests are added or updated (if applicable)
- [x] Documentation is updated (if needed)
- [x] Code is reviewed and approved

---

## 🛠️ Implementation Notes

Added tests to `utils_test.cpp`.

---

## 🔗 Related Issues / Links

[Issue #1138](https://github.com/rossvideo/Catena/issues/1138)

### CODE REVIEW

**Comment by ctwarog-ross** [comment]
Nice first code submission Keon, complete with unit tests. We can no longer hit the default case, so unsure what to do about that. Will get back

**Comment by abrown-RV** [comment]
> Nice first code submission Keon, complete with unit tests. We can no longer hit the default case, so unsure what to do about that. Will get back

We get Nathan's permission to put in an exclude block around it.

File: unittests/cpp/common/tests/utils_test.cpp [hunk 1]
```diff
@@ -225,8 +225,22 @@ TEST(UtilsTest, Value_Param_StructVariantArrayValues) {
     EXPECT_EQ(result, "[struct variant array values]");
 }
 
-TEST(UtilsTest, Value_Param_DefaultCase) {
+TEST(UtilsTest, Value_Param_StructArrayValues) {
+    st2138::Value value;
+    value.set_allocated_struct_array_values(new st2138::StructList());
```
Comment (abrown-RV) [inline unittests/cpp/common/tests/utils_test.cpp]:
```suggestion
    value.mutable_struct_array_values();
```
is the more "protobuf way" to do this, but I see you're just pattern matching. Can you update all the other instances as well?

**Review by abrown-RV — CHANGES_REQUESTED** [review]
very nice, we'll get that exclude direction for you

File: unittests/cpp/common/tests/utils_test.cpp (hunk 1 above)
Comment (KeonF-ross) [reply unittests/cpp/common/tests/utils_test.cpp]: I've updated most of the tests to use the `.mutable_xxx` methods. However, I wasn't able to find any for `int32`, `float32`, or `UndefinedValue`, so I left them the same. Was there some other method that you'd like me to use for those tests?

File: unittests/cpp/common/tests/utils_test.cpp [hunk 2]
```diff
@@ -160,14 +161,15 @@ TEST(UtilsTest, Value_Param_Float32) {
 
 TEST(UtilsTest, Value_Param_String) {
     st2138::Value value;
-    value.set_string_value("test string");
+    auto str = value.mutable_string_value();
+    str->assign("test string");
```
Comment (abrown-RV) [inline unittests/cpp/common/tests/utils_test.cpp]:
```suggestion
    value.set_string_value("test string");
```
Oh shoot, sorry for the confusion, its totally cool to do it this way with strings and other primitives.

File: unittests/cpp/common/tests/utils_test.cpp (hunk 1 above)
Comment (abrown-RV) [reply unittests/cpp/common/tests/utils_test.cpp]: the set_xxx functions are fine for primitives

**Comment by abrown-RV** [comment]
@KeonF-ross add //GCOVR_EXCL_START and //GCOVR_EXCL_END around the default and send Nathan a message with the filename and line numbers for the exclusion. There's other examples of this in the code you can match against.

**Review by abrown-RV — APPROVED** [review]
